### ISSUE TITLE
mss.mss() in while loop causes "Maximum number of clients reached Segmentation fault (core dumped)

### ISSUE THREAD

**Issue by jnguyen32** [BoboTiG/python-mss#72]
I'm on Ubuntu 18.04. This appears to be 100% reproducible with the following code:

```
while(True):
    sct = mss.mss()
```

Output error is:

`Maximum number of clients reachedSegmentation fault (core dumped)`

**Comment by BoboTiG** [comment]
Hello,

Thank you for the report.
Which MSS version?

**Comment by bachvtuan** [comment]
I have same issue with latest version 3.3.0 and I am on 16.04

**Comment by bachvtuan** [comment]
Downgrade to 3.2.1 and it works well.

**Comment by BoboTiG** [comment]
OK, thanks. I will have a look very soon.

**Comment by BoboTiG** [comment]
Fixed in the master branch. Sorry for the regression!

**Comment by BoboTiG** [comment]
MSS 3.3.1 published just now with the fix :)

**Comment by foone** [comment]
The fix is insufficient on python 2, the above code will still segfault with the same error, on MSS 3.3.1 or MSS 3.3.2. 
Downgrading to MSS 3.2.1, it'll work just fine. 
I inserted some debugging, and it turns out MSS.__del__ is never getting called. even adding an explicit "del sct" to the sample code doesn't work. 
The reason is that the MSS object has a bunch of circular references to it, thanks to the validate method inside _cfactory. Because it needs to access self.xlib and self.display, it has a reference to self, which causes MSS objects to never get garbage collected. 

I played around with fixing this but wasn't able to find a nice solution in the time I had.
Perhaps defining  __exit__ to call the self.xlib.XCloseDisplay method, so at least this pattern will work?

```
while True:
  with mss.MSS() as sct:
     pass
```
